Core/include/Acts/EventData/detail/TestSourceLink.hpp

```diff
@@ -89,11 +90,19 @@ struct TestSourceLink final {
   constexpr std::size_t index() const { return sourceId; }
 
   struct SurfaceAccessor {
-    const Acts::TrackingGeometry& trackingGeometry;
+    const Acts::TrackingGeometry* trackingGeometry = nullptr;
+    const Acts::Experimental::Detector* detector = nullptr;
```

Comment:
First, I like the change, I don't think we should store references around, I don't like this patter.
Then, see below we can probably get along with `SurfaceAccessor<typename geometry_t> and do not have dependency on `Experimental` directly.

Reply:
Let's also do an assert then that this is set ...

Reply:
Separated everything `TestSourceLink` related into #3445